### PR TITLE
[Added][BEL-3940] Widget usage in link creation

### PR DESCRIPTION
**Pull request by dom-techwriter**
:question: What
---

As part of [3940](https://belvoteam.atlassian.net/browse/BEL-3940), we wanted to add an additional usage section where we highlight the use of the widget. 

Note: I'll add the ruby reviewer as soon as I find the expert :)

:speech_balloon: Why
---

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,7 +33,34 @@ Or install it yourself as:
...
+```ruby
+require 'belvo'
+
+belvo = Belvo::Client.new(
+  'af6e69ff-43fa-4e10-8d90-3d217309a1e5',
+  'gdi64m68Lc6xUjIKN3aJF2fZd51wD36lTjGVyJO5xQBfL7PRsgFef-ADXBxIhUnd',
+  'sandbox'
+)
+
+id = resultFromCallbackFunction.link # The link_id you receive in your callback function
```
Comment (shekodn) [inline README.md:50]:
```suggestion
id = result_from_callback_function.link # The link_id you receive in your callback function
```

Comment (shekodn) [reply README.md]: in ruby we usually do snake_case. However, in JS we use camelCase

File: README.md [hunk 2]
```diff
@@ -33,19 +33,46 @@ Or install it yourself as:
...
+  'your-secret-id',
+  'your-secret-password',
+  'sandbox'
+)
+
+id = result_from_callback_function.link # The link_id you receive in your callback function
+
+  belvo.accounts.retrieve(link: new_link[id])
+
+  puts belvo.accounts.list
```
Comment (shekodn) [inline README.md:54]: What I would do:
-Take care of the indentation 
- I think we are missing a "begin" (we have a rescue below) 

image

File: README.md [hunk 3]
```diff
@@ -33,19 +33,46 @@ Or install it yourself as:
...
+```ruby
+require 'belvo'
+
+belvo = Belvo::Client.new(
+  'your-secret-id',
+  'your-secret-password',
+  'sandbox'
+)
+
+id = result_from_callback_function.link # The link_id you receive in your callback function
```
Comment (shekodn) [inline README.md:50]: why do we use this line? or where is `result_from_callback_function` declared?

Comment (shekodn) [reply README.md:50]: I know is the README, but this could cause confusion

Comment (dom-techwriter) [reply README.md:50]: All very good points!

So perhaps we don't need to use this line of code, however, I wanted to highlight that they can automatically retrieve the link_id after a successful widget link creation process. And in the widget, we usually ask them to have a callback function to handle the success message that contains the link ID. Do you have perhaps a better suggestion on how to illustrate this?

File: README.md (hunk 2 above)
Comment (dom-techwriter) [reply README.md:54]: Ahhh! See, that's why it's good to have an expert looking at this ;)

File: README.md (hunk 3 above)
Comment (shekodn) [reply README.md]: I know what you are trying to do. 

Remember that the widget is for the frontend side (javascript). Then, all the info is sent to the backend and this is when an SDK comes into play. I think we shouldn't mix frontend and backend code in the same snippet. 

IMO we could write about the thing you are trying to exemplify and then we can show with code snippets how to use the SDK and only use ruby code and backend logic in this snippet.

Comment (shekodn) [reply README.md]: remember that the widget is optional, and there might be a client that wants to use ruby, and maybe they are using their own widget/front-end implementation.

Comment (dom-techwriter) [reply README.md]: Ah, OK I think that makes sense :) But let me just have @midito comment on this as he's the one making the request for this documentation update.

Comment (midito) [reply README.md]: I don't have too much time to discuss on this.
I get Sergio's point by for now we don't have time to prepare an e2e example. It's on Dom's roadmap for next Q.
Customers/Devs will need to figure out and plan the solution by themselves but the approach on just using the SDKs to create a link is not going to work and it's not realistic.
Most of the customers are using the widget and this is the way to go. Creating links using the API is UNSECURE and something we should kill in the future or keep just for enterprise customers.
We need to always put the widget on first place to guide our integrators on using our best practices.

Comment (dom-techwriter) [reply README.md]: @shekodn I made a small change, and maybe for now let's see if it works and doesn't draw any confusion from our users? :)

File: README.md [hunk 4]
```diff
@@ -33,31 +33,60 @@ Or install it yourself as:
...
 )
 
 begin
-  new_link = belvo.links.register(
-    institution: 'banamex_mx_retail', 
-    username: 'janedoe', 
-    password: 'super-secret',
-    options: { access_mode: Belvo::Link::AccessMode::SINGLE }
-  )
+    # Get the link_id from the result of your widget callback function
```
Comment (shekodn) [inline README.md:51]: ✔️

**Review by shekodn — APPROVED** [review]
🎖️

**Pull request merged**